### PR TITLE
Ensure top-level components use the Redux store, if responsive

### PR DESCRIPTION
**Pull request by joshlory**
Prep for PR https://github.com/code-dot-org/code-dot-org/pull/19117 and https://github.com/code-dot-org/code-dot-org/pull/19399.  Anywhere we use `<ContentContainer>` should have a `<Provider>` with `responsive` and `isRtl` as registered reducers.

### CODE REVIEW

File: apps/src/sites/studio/pages/projects/public.js [hunk 1]
```diff
@@ -22,21 +22,21 @@ $(document).ready(() => {
...
-        <Provider store={getStore()}>
+      <Provider store={getStore()}>
+        <div>
+          <HeaderBanner
+            headingText={i18n.projects()}
+            subHeadingText={i18n.projectsSubHeading()}
+            short
+          />
+          <StartNewProject
+            isRtl={isRtl}
```
Comment (davidsbailey) [inline apps/src/sites/studio/pages/projects/public.js]: you may be able to remove isRtl here, now that you're supplying it via redux

Comment (joshlory) [reply apps/src/sites/studio/pages/projects/public.js]: Fixed in 5e787cb73438028d594ba3e387769280dc454938!

File: apps/test/unit/templates/studioHomepages/StudentHomepageTest.js [hunk 2]
```diff
@@ -101,7 +100,7 @@ describe('StudentHomepage', () => {
           isRtl={false}
```
Comment (Erin007) [inline apps/test/unit/templates/studioHomepages/StudentHomepageTest.js:100]: Still need to pass isRtl here?

Comment (joshlory) [reply apps/test/unit/templates/studioHomepages/StudentHomepageTest.js:100]: I think so.  There are still child components (like `StudentSections`) that don't pull this from Redux yet.

**Pull request merged**